src/coreclr-debug/activate.ts

```diff
@@ -7,126 +7,50 @@
 import * as fs from 'fs';
 import TelemetryReporter from 'vscode-extension-telemetry';
-import { CoreClrDebugUtil } from './util';
+import { CoreClrDebugUtil, DotnetInfo, DotNetCliError } from './util';
 import * as debugInstall from './install';
-import { PlatformInformation } from './../platform';
-import * as semver from 'semver';
+import * as path from 'path';
+import { Logger } from './../logger'
 
-const MINIMUM_SUPPORTED_DOTNET_CLI: string = '1.0.0-preview2-003121';
-
-let _reporter: TelemetryReporter = null;
-let _channel: vscode.OutputChannel = null;
 let _util: CoreClrDebugUtil = null;
+let _reporter: TelemetryReporter = null;
+let _logger: Logger = null;
 
-class DotnetInfo
-{
-    public Version: string;
-    public OsVersion: string;
-    public RuntimeId: string;
-}
-
-export function activate(context: vscode.ExtensionContext, reporter: TelemetryReporter) {
+export function activate(context: vscode.ExtensionContext, reporter: TelemetryReporter, logger: Logger) {
+    _util = new CoreClrDebugUtil(context.extensionPath, logger);
     _reporter = reporter;
-    _channel = vscode.window.createOutputChannel('coreclr-debug');
-    _util = new CoreClrDebugUtil(context.extensionPath, _channel);
-
-    if (CoreClrDebugUtil.existsSync(_util.installCompleteFilePath())) {
-        console.log('.NET Core Debugger tools already installed');
-        return;
-    }
-
-    checkForDotnetTools().then((dotnetInfo: DotnetInfo) => {
-        let installer = new debugInstall.DebugInstaller(_util);
-        _util.createInstallLog();
-
-        let statusBarMessage = vscode.window.setStatusBarMessage("Downloading and configuring the .NET Core Debugger...");
-
-        let runtimeId: string = '';
-        let installStage: string = "installBegin";
-        let installError: string = '';
-        let moreErrors: string = '';
-
-        getPlatformRuntimeId().then(rid => {
-            runtimeId = rid;
-        }).then(() => {
-            return writeInstallBeginFile();
-        }).then(() => {
-            return installer.install(runtimeId);
-        }).then(() => {
-            installStage = "completeSuccess";
-            statusBarMessage.dispose();
-            vscode.window.setStatusBarMessage('Successfully installed .NET Core Debugger.');
-        }).catch((error: debugInstall.InstallError) => {
-            const viewLogMessage = "View Log";
-            vscode.window.showErrorMessage('Error while installing .NET Core Debugger.', viewLogMessage).then(value => {
-                if (value === viewLogMessage) {
-                    _channel.show(vscode.ViewColumn.Three);
-                }
-            });
-            statusBarMessage.dispose();
-
-            installStage = error.installStage;
-            installError = error.errorMessage;
-            moreErrors = error.hasMoreErrors ? 'true' : 'false';
-        }).then(() => {
-            // log telemetry and delete install begin file
-            logTelemetry('Acquisition', {
```

Comment:
There is still an open question surrounding how we want to handle telemetry now that we have moved to the package manager.

---
In reply to: [86636070](https://github.com/OmniSharp/omnisharp-vscode/pull/873#discussion_r86636070) [](ancestors = 86636070)